test/unittests/execute_floating_point_test.cpp

```diff
@@ -367,6 +368,72 @@ TYPED_TEST(execute_floating_point_types, add)
     EXPECT_THAT(exec(TypeParam{0x0.287p2}, TypeParam{0x1.FFp4}), Result(TypeParam{0x1.048Ep5}));
 }
 
+TYPED_TEST(execute_floating_point_types, sub)
+{
+    using FP = FP<TypeParam>;
+    using Limits = typename FP::Limits;
+
+    auto instance = instantiate(parse(this->get_binop_code(Instr::f32_sub)));
+    const auto exec = [&](auto arg1, auto arg2) { return execute(*instance, 0, {arg1, arg2}); };
+
+    // fsub(+-inf, +-inf) = nan:canonical
+    EXPECT_THAT(exec(Limits::infinity(), Limits::infinity()), CanonicalNaN(TypeParam{}));
+    EXPECT_THAT(exec(-Limits::infinity(), -Limits::infinity()), CanonicalNaN(TypeParam{}));
+
+    // fsub(+-inf, -+inf) = +-inf
+    EXPECT_THAT(exec(Limits::infinity(), -Limits::infinity()), Result(Limits::infinity()));
+    EXPECT_THAT(exec(-Limits::infinity(), Limits::infinity()), Result(-Limits::infinity()));
+
+    // fsub(+-O, +-0) = +0
+    EXPECT_THAT(exec(TypeParam{0.0}, TypeParam{0.0}), Result(TypeParam{0.0}));
+    EXPECT_THAT(exec(-TypeParam{0.0}, -TypeParam{0.0}), Result(TypeParam{0.0}));
+
+    // fsub(+-0, -+0) = +-0
+    EXPECT_THAT(exec(TypeParam{0.0}, -TypeParam{0.0}), Result(TypeParam{0.0}));
+    EXPECT_THAT(exec(-TypeParam{0.0}, TypeParam{0.0}), Result(-TypeParam{0.0}));
+
+    // fsub(z1, +-0) = z1  (for z1 = +-inf)
+    EXPECT_THAT(exec(Limits::infinity(), TypeParam{0.0}), Result(Limits::infinity()));
+    EXPECT_THAT(exec(Limits::infinity(), -TypeParam{0.0}), Result(Limits::infinity()));
+    EXPECT_THAT(exec(-Limits::infinity(), TypeParam{0.0}), Result(-Limits::infinity()));
+    EXPECT_THAT(exec(-Limits::infinity(), -TypeParam{0.0}), Result(-Limits::infinity()));
+
+    for (const auto q : this->positive_special_values)
+    {
+        // fsub(z1, +-inf) = -+inf  (for z1 = +-q)
+        EXPECT_THAT(exec(q, Limits::infinity()), Result(-Limits::infinity()));
+        EXPECT_THAT(exec(q, -Limits::infinity()), Result(Limits::infinity()));
+        EXPECT_THAT(exec(-q, Limits::infinity()), Result(-Limits::infinity()));
+        EXPECT_THAT(exec(-q, -Limits::infinity()), Result(Limits::infinity()));
+
+        // fsub(+-inf, z2) = +-inf  (for z2 = +-q)
+        EXPECT_THAT(exec(Limits::infinity(), q), Result(Limits::infinity()));
+        EXPECT_THAT(exec(-Limits::infinity(), q), Result(-Limits::infinity()));
+        EXPECT_THAT(exec(Limits::infinity(), -q), Result(Limits::infinity()));
+        EXPECT_THAT(exec(-Limits::infinity(), -q), Result(-Limits::infinity()));
+
+        // fsub(z1, +-0) = z1  (for z1 = +-q)
+        EXPECT_THAT(exec(q, TypeParam{0.0}), Result(q));
+        EXPECT_THAT(exec(q, -TypeParam{0.0}), Result(q));
+        EXPECT_THAT(exec(-q, TypeParam{0.0}), Result(-q));
+        EXPECT_THAT(exec(-q, -TypeParam{0.0}), Result(-q));
+
+        // fsub(+-0, +-q2) = -+q2
+        EXPECT_THAT(exec(TypeParam{0.0}, q), Result(-q));
+        EXPECT_THAT(exec(-TypeParam{0.0}, -q), Result(q));
+
+        // Not part of the spec.
```

Comment:
What do you mean?

![image](https://user-images.githubusercontent.com/1863135/90002024-64a39700-dc92-11ea-9a9a-0d3a8b9d746c.png)

![image](https://user-images.githubusercontent.com/1863135/90002059-71c08600-dc92-11ea-979b-5bc4ebda2551.png)

Reply:
The `fsub(+-0, -q) = +-q` is missing.

Reply:
![image](https://user-images.githubusercontent.com/1863135/90002059-71c08600-dc92-11ea-979b-5bc4ebda2551.png)

This covers `fsub(+-0, -q2)` and shows that it negates the sign.

Reply:
You cannot resolve it like that because

> they must be resolved consistently: either the upper sign is chosen for all of them or the lower sign.

Reply:
Oh I see. You mean only these are defined:
```
fsub(+0, +q2) = -q2
fsub(-0, -q2) = +q2
```
But not these
```
fsub(-0, +q2) = -q2
fsub(+0, -q2) = +q2
```

Probably a bug in the spec, because text description doesn't mention any special cases, just "if z1 is zero"